snapcraft/plugins/_python/_pip.py

```diff
@@ -0,0 +1,263 @@
+        # Check to see if we have our own pip, yet. If not, we need to use the
+        # pip on the host (installed via build-packages) to grab our own.
+        try:
+            self._run([], stderr=subprocess.STDOUT)
+        except subprocess.CalledProcessError as e:
+            output = e.output.decode(sys.getfilesystemencoding()).strip()
+            if 'no module named pip' in output.lower():
+                logger.info('Fetching pip, setuptools, and wheel...')
+                env = {'PYTHONHOME': '/usr'}
+
+                # Using the host's pip, install our own pip and other tools we
+                # need.
+                self.download({'pip', 'setuptools', 'wheel'}, env=env)
+                self.install({'pip', 'setuptools', 'wheel'}, env=env)
+            else:
+                raise e
+
+    def download(self, args, **kwargs):
+        """Using the requested args, download packages into a cache.
+
+        Simple wrapper for `pip download`. Args are appended at the end, and
+        any keyword args are passed along to `common.run_output`. Note that
+        this function does not install the fetched packages.
+
+        :param list args: Arguments for `pip download`. Package names, for
+                          example, or the path to a dir containing a setup.py.
+        """
+        if not args:
+            return  # No packages to download
+
+        # Using pip with a few special parameters:
+        #
+        # --disable-pip-version-check: Don't whine if pip is out-of-date with
+        #                              the version on pypi.
+        # --dest: Download packages into the directory we've set aside for it.
+        self._run(['download', '--disable-pip-version-check', '--dest',
+                   self._python_package_dir] + list(args), **kwargs)
+
+    def install(self, args, **kwargs):
+        """Using the requested args, install packages from the cache.
+
+        Simple wrapper for `pip install`. Args are appended at the end, and
+        any keyword args are passed along to `common.run_output`. Note that
+        the packages should have already been downloaded via `self.download`.
+
+        :param list args: Arguments for `pip install`. Package names, for
+                          example, or the path to a dir containing a setup.py.
+        """
+        if not args:
+            return  # No packages to install
+
+        # Using pip with a few special parameters:
+        #
+        # --user: Install packages to PYTHONUSERBASE, which we've pointed to
+        #         the installdir.
+        # --no-compile: Don't compile .pyc files. FIXME: Not sure why
+        # --no-index: Don't hit pypi, assume the packages are already
+        #             downloaded (i.e. by using `self.download()`)
+        # --find-links: Provide the directory into which the packages should
+        #               have already been fetched
+        # --ignore-installed: If the package is already installed, reinstall
```

Comment:
this is wrong, we only use it to install the latest `pip`, `wheels` and `setuptools`

```python
        args = ['pip', 'setuptools', 'wheel']

        pip_command = [self._get_python_command(), '-m', 'pip']

        pip = _Pip(exec_func=subprocess.check_call,
                   runnable=pip_command,
                   package_dir=self._python_package_dir, env=env,
                   extra_install_args=['--ignore-installed'])
```

And this is beacuse we are using python on the host to install these tools in the """venv"""

Reply:
Ah! Good catch, that makes more sense indeed.

Reply:
Fixed. `--ignore-installed` is now only used when pip etc. are being installed.